pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -383,10 +417,17 @@ func getPackageDetailsFromCBS(CBSRefName string) (pkgShortName, pkgName, pkgVers
 	return
 }
 
-func getClusterBootstrapTemplate(ctx context.Context, k8sClient client.Client, tkrName string) *runtanzuv1alpha3.ClusterBootstrapTemplate {
+func getClusterBootstrapTemplate(ctx context.Context, k8sClient client.Client, tkrName string, infrastructureName string) *runtanzuv1alpha3.ClusterBootstrapTemplate {
+	var objKey client.ObjectKey
+
 	clusterBootstrapTemplate := &runtanzuv1alpha3.ClusterBootstrapTemplate{}
-	objKey := client.ObjectKey{Name: tkrName, Namespace: constants.TkgNamespace}
+	if infrastructureName == "tkgs" {
+		objKey = client.ObjectKey{Name: tkrName, Namespace: constants.TKGSClusterClassNamespace}
+	} else {
+		objKey = client.ObjectKey{Name: tkrName, Namespace: constants.TkgNamespace}
+	}
```

Comment:
wondering if you could just pass ```systemNamespace``` to this function as a parameter instead of having this ```if,else``` block to determine the namespace

Reply:
Fixed